homeassistant/components/influxdb/__init__.py

```diff
@@ -55,98 +65,133 @@
 BATCH_TIMEOUT = 1
 BATCH_BUFFER_SIZE = 100
 
-COMPONENT_CONFIG_SCHEMA_ENTRY = vol.Schema(
-    {vol.Optional(CONF_OVERRIDE_MEASUREMENT): cv.string}
+DB_CONNECTION_FAILURE_MSG = ()
+
+COMPONENT_CONFIG_SCHEMA_CONNECTION = {
+    # Connection config for V1 API.
+    vol.Optional(CONF_HOST): cv.string,
+    vol.Inclusive(CONF_USERNAME, "authentication"): cv.string,
+    vol.Inclusive(CONF_PASSWORD, "authentication"): cv.string,
+    vol.Optional(CONF_DB_NAME, default=DEFAULT_DATABASE): cv.string,
+    vol.Optional(CONF_PATH): cv.string,
+    vol.Optional(CONF_PORT): cv.port,
+    vol.Optional(CONF_SSL): cv.boolean,
+    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
+    # Connection config for V2 API.
+    vol.Inclusive(CONF_API_V2, "v2_api_config"): True,
+    vol.Optional(CONF_URL, default=DEFAULT_URL): cv.url,
```

Comment:
V1, works with an URL as well, which is constructed by the existing options. Can we merge this behavior? There is no need to have everything twice.

Reply:
The main issue I had here were the defaults since they are already specified for those fields based on what V1 wanted.  I wanted V2 to default to using the cloud service URL. But even if I continue to default `host` to `localhost` for V2 then the port should still change to `9999` as that is the default port for a local v2 deployment.

The only other issue was `verify_ssl` is not a configuration option in the v2.  You cannot tell the V2 API to skip verifying the SSL certificate, you must have a valid certificate if you're going to use SSL.

And I guess besides that it just felt cleaner to me to match what the V2 docs were telling people to do.  All the Influx docs say put in URL, Bucket and Token.  We would be forcing users to break apart their URL into pieces to fill out our config.

If you disagree though I will change it.  In documentation I will probably have to note in the descriptions of these fields when the default is different for V1 vs. V2 instead of making full use of the "default" option in the configuration object.

Reply:
The integration either defaults to v1 or v2, which is expected to match the default values. Another option could be setting the default value based on the API version selected.

Reply:
Ok adjusted for this feedback.  Now assembling the URL for V2 from the `ssl`, `host`, `port` and `path` config options and removed the `url` option.  Will wait for the second opinion on how to handle the `api_v2` config option.

Going to go adjust the doc PR accordingly.